indico/modules/designer/controllers.py

```diff
@@ -281,9 +296,15 @@ def _process_GET(self):
     def _process_POST(self):
         data = dict({'background_position': 'stretch', 'items': []}, **request.json['template'])
         self.validate_json(TEMPLATE_DATA_JSON_SCHEMA, data)
-        invalid_placeholders = {x['type'] for x in data['items']} - set(get_placeholder_options()) - {'fixed'}
+        invalid_placeholders = {x['type'] for x in data['items']} - set(get_placeholder_options())
         if invalid_placeholders:
             raise UserValueError('Invalid item types: {}'.format(', '.join(invalid_placeholders)))
+        image_items = [item for item in data['items'] if item['type'] == 'fixed_image']
+        for image_item in image_items:
+            if image_item.get('image_id'):
+                image = DesignerImageFile.get(image_item['image_id'])
```

Comment:
Instead of querying the images on by one, why not build a set with all the images of the template (`{img.id for img in self.template.images}`) and then simply check if every `image_id` is in that set?